### PR TITLE
ddl: handle the incorrect number of placement followers

### LINKED ISSUE #30453: FOLLOWERS=99 should be illegal
The following syntax is currently considered legal in TiDB:
```sql
CREATE PLACEMENT policy p1 CONSTRAINTS="[+region=us-east-2]" FOLLOWERS=99;
create table t1 (a int) followers=99;
```

It seems that PD does not validate this either, but I propose the following:
- A non even number of followers returns a warning (raft recommends using an even number of followers, so the total group has an odd number).
- A number of followers greater than 8 returns an error.

### PR DESCRIPTION
**Pull request by sylzd**
### What problem does this PR solve?

Issue Number: close #30453 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Limit the maximum size of follower numbers to 8 in placement policies.
```

### CODE REVIEW

**Comment by sylzd** [comment]
@morgo  I am not sure if we should return warning after ddl for these reasons:
1. It seems all ddl warning info is returned before ddl executing, and the ddl will not execute actually, such as `drop if exists`、 `create if not exists`.   
2. some cases we need odd followers, like `follower read`, `geo partition`, maybe some followers need not participate in vote ...
3. the check function is used somewhere without session.🤣

**Comment by morgo** [comment]
@sylzd good points. Let me take a look and make sure we handle these cases correctly.

**Review by djshow832 — APPROVED** [review]
Please add a release note to the PR description. Rest LGTM

File: ddl/placement/bundle.go [hunk 1]
```diff
@@ -207,6 +207,10 @@ func newBundleFromOptions(options *model.PlacementSettings) (bundle *Bundle, err
 		return nil, fmt.Errorf("%w: options can not be nil", ErrInvalidPlacementOptions)
 	}
 
+	if options.Followers > uint64(8) {
```
Comment (AilinKid) [inline ddl/placement/bundle.go:210]: missed this? `A non even number of followers returns a warning`

**Comment by AilinKid** [comment]
> @morgo I am not sure if we should return warning after ddl for these reasons:
> 
> 1. the check function `checkPolicyValidation` is used somewhere without session for asynchronous executing such as `onCreatePlacementPolicy`, we need much redundant code to handle the case.rofl
> 2. some cases we need odd followers, like `follower read`, `geo partition`, and some followers need not participate in vote ...
> 3. It seems all ddl warning info is returned before ddl executing, and the ddl will not execute actually, such as `drop if exists`、 `create if not exists`.

I'm not sure about 2.
but for 1 & 3, ddl background warnings can be fetched into `DDLReorgMeta` which is designed for reorg type job, maybe we can utilize it if necessary.

**Comment by sylzd** [comment]
> > @morgo I am not sure if we should return warning after ddl for these reasons:
> > 
> > 1. the check function `checkPolicyValidation` is used somewhere without session for asynchronous executing such as `onCreatePlacementPolicy`, we need much redundant code to handle the case.rofl
> > 2. some cases we need odd followers, like `follower read`, `geo partition`, and some followers need not participate in vote ...
> > 3. It seems all ddl warning info is returned before ddl executing, and the ddl will not execute actually, such as `drop if exists`、 `create if not exists`.
> 
> I'm not sure about 2... but for 1 & 3, ddl background warnings can be fetched into `DDLReorgMeta` which is designed for reorg type job, maybe we can utilize it if necessary.

LGTM~ that helps~ I will take the advice from you if we are sure to append a warning here. but I still have some questions: 🤣
1. How do we generally define the error code that does not exist in MySQL?  like `8XXX`?
2. how can we avoid duplicated warning append? we check the validation twice, and I do not find a similar case about warnings.

**Comment by morgo** [comment]
> 1. How do we generally define the error code that does not exist in MySQL?  like `8XXX`?

Yes exactly. MySQL error codes are far from this range, so it is safe to use.

> 2. how can we avoid duplicated warning append? we check the validation twice, and I do not find a similar case about warnings.

There is no builtin support. You can use `vars.StmtCtx.GetWarnings()` to get the warnings and `vars.StmtCtx.SetWarnings(warns)` to set the warnings if you need to implement a function for this.

**Comment by morgo** [comment]
For (2) we might need to skip this rule for now. It is fine without it. This PR LGTM.

**Review by morgo — APPROVED** [review]

**Pull request merged**